sdk/eventhub/azure-eventhub/azure/eventhub/_tracing.py

```diff
@@ -0,0 +1,270 @@
+
+                if additional_attributes:
+                    for key, value in additional_attributes.items():
+                        if value is not None:
+                            message_span.add_attribute(key, value)
+
+    except Exception as exp:  # pylint:disable=broad-except
+        _LOGGER.warning("trace_message had an exception %r", exp)
+
+    return message
+
+
+def get_span_links_from_received_events(events: Union[EventData, Iterable[EventData]]) -> List[Link]:
+    """Create span links from received events.
+
+    This will extract the traceparent and tracestate from the event properties and create span links
+    based on these values. The time the event was enqueued is also added as a link attribute.
+    """
+    # pylint:disable=isinstance-second-argument-not-valid-type
+    trace_events = events if isinstance(events, Iterable) else (events,)
+    links = []
+    try:
+        for event in trace_events:
+            if event.properties:
+                headers = {}
+                traceparent = (event.properties.get(TRACE_PARENT_PROPERTY, b"") or
+                               event.properties.get(TRACE_DIAGNOSTIC_ID_PROPERTY, b""))
+                if hasattr(traceparent, "decode"):
+                    traceparent = traceparent.decode(TRACE_PROPERTY_ENCODING)
+                if traceparent:
+                    headers["traceparent"] = traceparent
+
+                tracestate = event.properties.get(TRACE_STATE_PROPERTY, b"")
+                if hasattr(tracestate, "decode"):
+                    tracestate = tracestate.decode(TRACE_PROPERTY_ENCODING)
+                if tracestate:
+                    headers["tracestate"] = tracestate
+
+                enqueued_time = event.system_properties.get(PROP_TIMESTAMP)
+                attributes = {'enqueuedTime': enqueued_time} if enqueued_time else None
+                links.append(Link(headers, attributes=attributes))
+    except AttributeError:
+        pass
+    return links
+
+
+def get_span_links_from_batch(batch: EventDataBatch) -> List[Link]:
+    """Create span links from a batch of events."""
+    links = []
+    try:
+        for event in batch._internal_events:  # pylint: disable=protected-access
+            # pylint: disable=protected-access
+            msg = event if isinstance(event, AmqpAnnotatedMessage) else event._message
+            link = get_span_link_from_message(msg)
+            if link:
+                links.append(link)
+    except AttributeError:
+        pass
+    return links
+
+
+def get_span_link_from_message(message: Union[AmqpAnnotatedMessage, Message]) -> Optional[Link]:
+    """Create a span link from a message.
+
+    This will extract the traceparent and tracestate from the message application properties and create span links
+    based on these values.
+    """
+    headers = {}
+    try:
+        if message.application_properties:
```

Comment:
same here, we should make sure to add link even if there are no properties on the message